### PR TITLE
Removed deprecated shouldBeEqualToComparingFields methods

### PR DESCRIPTION
**Pull request by sksamuel**

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt [hunk 1]
```diff
@@ -299,42 +299,6 @@ fun <T : Any> beEqualToIgnoringFields(
...
- * will be same as that we get with shouldBe matcher.
- *
- * @param other the other class to which equality need to be checked.
- * @param ignorePrivateFields specify whether private fields should be ignored in equality check or not, default value is true
- * @param ignoreComputedFields specify whether computed fields should be ignored in equality check or not, default value is true
- * @param useDefaultShouldBeForFields fully qualified names of data type for which we need to use default shouldBe, default empty list.
- *
- * */
-@Deprecated(
-   message = "This will be removed in Kotest 7.0",
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt:319]: Nit:
I mean, we did promise to keep it until Kotest 7.0, not 6.0

Comment (LeoColman) [reply kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt:319]: (applies to most of the removed assertions)

File: kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/async/timeout.jvm.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ import kotlin.time.toKotlinDuration
  * Assert [thunk] does not complete within [duration].
  */
 @Deprecated(
-   "Updated to use Kotlin time types. This function will be removed in 6.0.",
+   "Updated to use Kotlin time types. This function will be removed in 6.1.",
```
Comment (LeoColman) [inline kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/async/timeout.jvm.kt:12]: But the 6.0 guys get a kick to 6.1.

Poor 7.0 deprecations :(

**Review by LeoColman — APPROVED** [review]

File: kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt (hunk 1 above)
Comment (sksamuel) [reply kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt:319]: But that was a typo I think becuse they were deprecated in 2022

File: kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/async/timeout.jvm.kt (hunk 2 above)
Comment (sksamuel) [reply kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/async/timeout.jvm.kt:12]: Ha. I was basing it on time deprecated

**Pull request merged**